packages/contracts/contracts/optimistic-ethereum/OVM/accounts/OVM_ECDSAContractAccount.sol

```diff
@@ -69,6 +71,22 @@ contract OVM_ECDSAContractAccount is iOVM_ECDSAContractAccount {
             Lib_ExecutionManagerWrapper.ovmCHAINID()
         );
 
+        // Need to make sure that the gas is sufficient to execute the transaction.
+        uint256 intrinsicGas = SafeMath.mul(_encodedTransaction.length, 16);
+        uint256 gasLimit = SafeMath.mul((transaction.gasLimit % 10000), 10000);
+        uint256 gasLeft = gasleft();
+        require(
+            gasLeft >= gasLimit,
+            string(
```

Comment:
This was helpful for debugging - too verbose to leave in?

Reply:
This pattern isn't found elsewhere in the contracts so I'd imagine we should remove it but curious as to why most revert messages are static in solidity - just a gas thing?